### ISSUE TITLE
Fix S2857 FP: Rule is not checking SQL keywords in const interepolated string

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6633]
### Description

S2857 fail to check for SQL keywords inside a:
- `constant` interpolated string:
- interpolated string with `nameof` expression

```cs
using System.Data.SqlClient;

public void MyMethod(string unknownValue)
{
  const string One = "One";
  const string myString = $"{One}Two"; // Noncompliant FP
  const string myString1 = $"{nameof(One)}Two"; // Noncompliant FP
  string myString2 = $"{nameof(unknownValue)}Two"; // Noncompliant FP
}
```

**Comment by martin-strecker-sonarsource** [comment]
see also #6126

**Comment by cristian-ambrosini-sonarsource** [comment]
As reported by #8102, non-const string are also raising FP:
```cs
void Example(string parameter)
{
    string nonConstOne = "One";
    string empty = string.Empty;

    string s1 = $"{nonConstOne}";              // Compliant
    string s2 = $"{nonConstOne}Two";           // Noncompliant FP
    string s3 = $"{empty}";                    // Compliant
    string s4 = $"{empty}Two";                 // Compliant

    string s5 = $"{{{nonConstOne}}}";          // Noncompliant FP
                                                // Noncompliant@-1 FP
}
```
